src/Donors/Controllers/DonorRequestController.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Give\Donors\Controllers;
+
+use Give\Donations\ValueObjects\DonationMetaKeys;
+use Give\Donors\Models\Donor;
+use Give\Donors\ValueObjects\DonorRoute;
+use Give\Framework\QueryBuilder\JoinQueryBuilder;
+use Give\Framework\QueryBuilder\QueryBuilder;
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Response;
+
+/**
+ * @unreleased
+ */
+class DonorRequestController
+{
+    /**
+     * @unreleased
+     *
+     * @return WP_Error | WP_REST_Response
+     */
+    public function getDonor(WP_REST_Request $request)
+    {
+        $donor = Donor::find($request->get_param('id'));
+
+        if ( ! $donor) {
+            return new WP_Error('donor_not_found', __('Donor not found', 'give'), ['status' => 404]);
+        }
+
+        return new WP_REST_Response($this->escDonor($donor));
+    }
+
+    /**
+     * @unreleased
+     */
+    public function getDonors(WP_REST_Request $request): WP_REST_Response
+    {
+        $page = $request->get_param('page');
+        $perPage = $request->get_param('per_page');
+        $sortColumn = $this->getSortColumn($request->get_param('sort'));
+        $sortDirection = $request->get_param('direction');
+        $mode = $request->get_param('mode');
+
+        $query = Donor::query();
+
+        // Donors only can be donors if they have donations associated with them
+        if ($request->get_param('onlyWithDonations')) {
+            $query->join(function (JoinQueryBuilder $builder) use ($mode) {
+                // The donationmeta1.donation_id should be used in other "donationmeta" joins to make sure we are retrieving data from the proper donation
+                $builder->innerJoin('give_donationmeta', 'donationmeta1')
+                    ->joinRaw("ON donationmeta1.meta_key = '" . DonationMetaKeys::DONOR_ID . "' AND donationmeta1.meta_value = ID");
+
+                // Include only current payment "mode"
+                $builder->innerJoin('give_donationmeta', 'donationmeta2')
+                    ->joinRaw("ON donationmeta2.meta_key = '" . DonationMetaKeys::MODE . "' AND donationmeta2.meta_value = '{$mode}' AND donationmeta2.donation_id = donationmeta1.donation_id");
+            });
```

Comment:
@glaubersilva @jonwaldstein 
Thinking ahead to database changes, I _strongly_ recommend that we update (or create) a `DonorQueryBuilder` that has declarative methods for things like this. So this craziness becomes:
```php
$query->whereDonorsHaveDonations();
```
It's cool for the REST APIs to use the query builder, but we want to reduce modifications that are directly coupled to the database structure (such as referencing meta like we are here).

Reply:
@JasonTheAdams If you don't mind, I would like to refactor this in a subsequent PR because these endpoints will not be immediately used in the codebase, but I'm afraid to spend too much time on it and delay the merge into the epic even more - as we are changing the namespace of the campaign routes, we need to merge it before the public release. 

Also, I would like to let the folder structure changes mentioned by @jonwaldstein  [**here**](https://github.com/impress-org/givewp/pull/7699#issuecomment-2667062607) be included in this subsequent PR as well.

Reply:
So long as the API itself is finalized, I'm cool with moving under-the-hood changes to a subsequent PR. BUT! You must agree to actually **plan** to do it. None of this, "oh yeah, we'll do that later for sure" stuff. 😁

Reply:
Sure! We can do it later. 

![TrueTrollGIF](https://github.com/user-attachments/assets/4797f694-e1c5-4385-b7fb-ce6b750411c0)

Ok, I'm kidding! 😅 

My plan is to start a new PR as soon the Campaigns tasks (related to the public release) pile gets down, so probably on the next fun Friday I'll already open a PR for it.